config/nvidia-flags

```diff
@@ -0,0 +1,122 @@
+# help@hdfgroup.org.
+
+
+# This file should be sourced into configure if the compiler is the
+# NVIDIA nvc compiler or a derivative.  It is careful not to do anything
+# if the compiler is not nvcc; otherwise `cc_flags_set' is set to `yes'
+#
+
+# Get the compiler version in a way that works for nvc
+# unless a compiler version is already known
+#
+#   cc_vendor:    The compiler name: nvc
+#   cc_version:   Version number: 5.0-2, 5.2-2
+#
+if test X = "X$cc_flags_set"; then
+    cc_version="`$CC $CFLAGS -V 2>&1 |grep '^nvc '`"
+    if test X != "X$cc_version"; then
+        is_mpi="`$CC $CFLAGS -help 2>&1 |grep 'MPI'`"
+        cc_vendor=`echo $cc_version |sed 's/\([a-z]*\).*/\1/'`
+        cc_version=`echo $cc_version |sed 's/nvc \([-a-z0-9\.\-]*\).*/\1/'`
+        echo "compiler '$CC' is NVIDIA $cc_vendor-$cc_version"
+
+        # Some version numbers
+        # NVIDIA version numbers are of the form: "major.minor-patch"
+        cc_vers_major=`echo $cc_version | cut -f1 -d.`
+        cc_vers_minor=`echo $cc_version | cut -f2 -d. | cut -f1 -d-`
+        cc_vers_patch=`echo $cc_version | cut -f2 -d. | cut -f2 -d-`
+        test -n "$cc_vers_major" || cc_vers_major=0
+        test -n "$cc_vers_minor" || cc_vers_minor=0
+        test -n "$cc_vers_patch" || cc_vers_patch=0
+        cc_vers_all=`expr $cc_vers_major '*' 1000000 + $cc_vers_minor '*' 1000 + $cc_vers_patch`
+    fi
+fi
+
+# Common PGI flags for various situations
+if test "X-nvc" = "X-$cc_vendor" -o "X-nvcc" = "X-$cc_vendor"; then
+    # Insert section about version specific problems from compiler flags here,
+    # if necessary.
+
+    arch=
+    # Architecture-specific flags
+    # Nothing currently. (Uncomment code below and modify to add any)
+    #case "$host_os-$host_cpu" in
+    #    *-i686)
+    #        arch="-march=i686"
+    #        ;;
+    #esac
+
+    # Host-specific flags
+    # Nothing currently. (Uncomment code below and modify to add any)
+    #case "`hostname`" in
+    #    sleipnir.ncsa.uiuc.edu)
+    #        arch="$arch -pipe"
+    #        ;;
+    #esac
+
+    ###########
+    # General #
+    ###########
+
+    # Default to C99 standard.
+    H5_CFLAGS="$H5_CFLAGS $arch -c99 -Minform=warn"
+
+    ##############
+    # Production #
+    ##############
+
+    # NDEBUG is handled explicitly by the configure script
+    PROD_CFLAGS="-fast"
+
+    #########
+    # Debug #
+    #########
+
+    # NDEBUG is handled explicitly by the configure script
+    # -g is handled by the symbols flags
+    DEBUG_CFLAGS="-Mbounds"
+
+    ###########
+    # Symbols #
+    ###########
+
+    NO_SYMBOLS_CFLAGS="-s"
+    SYMBOLS_CFLAGS="-g"
+
+    #############
+    # Profiling #
+    #############
+
+    PROFILE_CFLAGS="-Mprof=func,line"
+    # Use this for profiling with gprof
+    #PROFILE_CFLAGS="-pg"
+
+    ################
+    # Optimization #
+    ################
+
+    HIGH_OPT_CFLAGS="-O4"
+    DEBUG_OPT_CFLAGS="-gopt -O2"
+    NO_OPT_CFLAGS="-O0"
```

Comment:
Change HIGH and DEBUG to -O1 and comment that -O2+ currently has test failures.

Reply:
IIRC, debug builds work fine. It was specifically when -O2 or greater appears alongside NDEBUG being defined